File: SWEA/D2/1940. 가랏！ RC카！/가랏！ RC카！.java
import java.util.Scanner;

public class Solution {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int T = sc.nextInt();
		for(int tc=1; tc<=T; tc++) {
			int N = sc.nextInt();
			
			int distance = 0;
			int speed = 0;
			
			for(int n=0; n<N; n++) {
				int c1 = sc.nextInt();
				
				if(c1 == 1) {
					int c2 = sc.nextInt();
					speed += c2;
				}
				else if(c1 == 2) {
					int c2 = sc.nextInt();
					speed -= c2;
					
					if(speed < 0) {
						speed = 0;
					}
				}
				
				distance += speed;
			}
			
			System.out.printf("#%d %d\n", tc, distance);
		}
	}

}

File: SWEA/D2/1940. 가랏！ RC카！/가랏！ RC카！.py
T = int(input())

for tc in range(1, T+1):
    N = int(input())

    distance = 0
    speed = 0
    for n in range(N):
        c = list(map(int, input().split()))

        if c[0] == 1:  # 가속
            speed += c[1]

        elif c[0] == 2:  # 감속
            speed -= c[1]
            if speed < 0:
                speed = 0

        distance += speed

    print(f'#{tc} {distance}')
